### PR TITLE
[vcpkg baseline][stdexec] Fix compilation failure under Windows

### PR DESCRIPTION
**Pull request by jimwang118**
Detected by https://dev.azure.com/vcpkg/public/_build/results?buildId=104733&view=results
```
CMake Error at CMakeLists.txt:44 (project):
  VERSION "0.10.a.0" format invalid.
```
Fix the regular expression to get the correct version number.
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test pass with following triplets:

```
x64-windows
x64-windows-static
```

### CODE REVIEW

**Comment by dg0yt** [comment]
The vcpkg port downloads  
https://raw.githubusercontent.com/brycelelbach/wg21_p2300_execution/R7/execution.bs  
which has
~~~
Revision: 8
~~~
so it doesn't need this PR.

But I don't know if all downloads are done consistently in this port.

**Comment by jimwang118** [comment]
> The vcpkg port downloads https://raw.githubusercontent.com/brycelelbach/wg21_p2300_execution/R7/execution.bs which has
> 
> ```
> Revision: 8
> ```
> 
> so it doesn't need this PR.
> 
> But I don't know if all downloads are done consistently in this port.

Yes, the version downloaded in vcpkg is 8, but now after the update, the download path is added upstream in the new cmakelists.txt: "https://raw.githubusercontent.com/cplusplus/sender-receiver/main/execution.bs".
The version number obtained in the updated download is 10.a, so this version number is not recognized in the project version in cmake during compilation.

**Review by MonicaLiu0311 — DISMISSED** [review]

**Review by dg0yt — CHANGES_REQUESTED** [review]
All downloads must be handled in the portfile, not in CMakeLists.txt.
Or asset caching and offline builds break.

**Comment by dg0yt** [comment]
> All downloads must be handled in the portfile, not in CMakeLists.txt.

Still not implemented correctly. Is it so hard?

File: ports/stdexec/portfile.cmake [hunk 1]
```diff
@@ -23,9 +25,9 @@ vcpkg_download_distfile(RAPIDS_cmake
 file(COPY "${RAPIDS_cmake}" DESTINATION "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/")
 
 vcpkg_download_distfile(execution_bs
-    URLS "https://raw.githubusercontent.com/brycelelbach/wg21_p2300_execution/R7/execution.bs"
+    URLS "https://raw.githubusercontent.com/cplusplus/sender-receiver/main/execution.bs"
```
Comment (dg0yt) [inline ports/stdexec/portfile.cmake:28]: :clock9: ... :clock10: ... :clock11: ...

File: ports/stdexec/portfile.cmake [hunk 2]
```diff
@@ -23,9 +25,9 @@ vcpkg_download_distfile(RAPIDS_cmake
...
     FILENAME "execution.bs"
-    SHA512 90f2a1d150b03c29bb05a5420e091c2371cb973335a089916716d778bc1081764436dc1ff0fec60f642ddb0ca5492c8b0c3a6d5451c2d60a42911f918fe980fa
+    SHA512 90bb992356f22e4091ed35ca922f6a0143abd748499985553c0660eaf49f88d031a8f900addb6b4cf9a39ac8d1ab7c858b79677e2459136a640b2c52afe3dd23
```
Comment (dg0yt) [inline ports/stdexec/portfile.cmake:30]: ... :boom:

**Review by dg0yt — COMMENTED** [review]
In addition:

File: ports/stdexec/portfile.cmake (hunk 1 above)
Comment (jimwang118) [reply ports/stdexec/portfile.cmake:28]: The download address has been modified.

**Pull request merged**